lib/src/client_offline_mixin.dart

```diff
@@ -0,0 +1,662 @@
+              await updateAccessedAt(txn, store.name, record.key);
+            }
+          });
+          final count = await store.count(db);
+          return Response(data: {
+            'total': count,
+            cacheResponseContainerKey: records.map((record) {
+              final map = Map<String, dynamic>();
+              record.value.entries.forEach((entry) {
+                map[entry.key] = entry.value;
+              });
+              return map;
+            }).toList(),
+          });
+        }
+        throw AppwriteException(
+          "Client is offline and data is not cached",
+          0,
+          "general_offline",
+        );
+      case HttpMethod.post:
+      case HttpMethod.patch:
+      case HttpMethod.put:
+      case HttpMethod.delete:
+        switch (method) {
+          case HttpMethod.post:
+            if (params.containsKey('data')) {
+              final documentId = params['documentId'];
+              cacheKey = documentId;
+              final document = Map<String, dynamic>.from(params['data']);
+              document['\$createdAt'] =
+                  DateTime.now().toUtc().toIso8601String();
+              document['\$updatedAt'] =
+                  DateTime.now().toUtc().toIso8601String();
+              document['\$id'] = documentId;
+              document['\$collectionId'] = pathSegments[4];
+              document['\$databaseId'] = pathSegments[2];
+              document['\$permissions'] = params['permissions'] ?? [];
+              await db.transaction((txn) async {
+                await upsertCache(txn, store, document, key: cacheKey);
+                queuedWriteKey = await addQueuedWrite(
+                  txn,
+                  method,
+                  path,
+                  headers,
+                  params,
+                  cacheModel,
+                  cacheKey,
+                  cacheResponseIdKey,
+                  cacheResponseContainerKey,
+                  null,
+                );
+              });
+            }
+            break;
+          case HttpMethod.delete:
+            if (cacheKey.isNotEmpty) {
+              await db.transaction((txn) async {
+                final previous = await store.record(cacheKey).get(txn);
+                await deleteCache(txn, store, key: cacheKey);
+                queuedWriteKey = await addQueuedWrite(
+                  txn,
+                  method,
+                  path,
+                  headers,
+                  params,
+                  cacheModel,
+                  cacheKey,
+                  cacheResponseIdKey,
+                  cacheResponseContainerKey,
+                  previous,
+                );
+              });
+            }
+            break;
+          case HttpMethod.put:
+          case HttpMethod.patch:
+            final entry = Map<String, dynamic>();
+            if (params.containsKey('data')) {
+              entry.addAll(Map<String, dynamic>.from(params['data']));
```

Comment:
Here we map the `data` attributes to the top level of the map before inserting in the cache, but we don't re-nest it when pulling it out that I can see, will that cause issues when mapping to the `Document` response model?

Reply:
This `entry` gets added into a local collection and needs to match the data that is returned by the API call.

The data that ends up being sent to the server comes from a different local collection, the queue of write operations, which has a different schema.